src/runway.rs

```diff
@@ -440,10 +490,77 @@ where
         self.send_consensus_notification(NotificationIn::UnitParents(u_hash, p_hashes));
     }
 
+    fn on_newest_response(
+        &mut self,
+        unchecked_response: UncheckedSigned<NewestUnitResponse<H, D, MK::Signature>, MK::Signature>,
+    ) {
+        let starting_round_sender = match self.starting_round_sender.take() {
+            Some(sender) => sender,
+            None => {
+                log::debug!(target: "AlephBFT-member", "Starting round already sent, ignoring newest unit response");
+                return;
+            }
+        };
+        let response = match unchecked_response.check(self.keybox) {
+            Ok(checked) => checked.into_signable(),
+            Err(e) => {
+                log::debug!(target: "AlephBFT-member", "incorrectly signed response: {:?}", e);
+                return;
+            }
+        };
+
+        if response.salt != self.salt {
+            debug!(target: "AlephBFT-member", "Ignoring newest unit response with an unknown salt: {:?}", response);
+            return;
+        }
+
+        if let Some(unchecked_unit) = response.unit {
+            if unchecked_unit.as_signable().creator() != self.index() {
+                log::debug!(target: "AlephBFT-member", "Not our unit in a response:  {:?}", unchecked_unit.into_signable());
+                return;
+            }
+            let checked_unit = match unchecked_unit.check(self.keybox) {
+                Ok(checked) => checked,
+                Err(e) => {
+                    log::debug!(target: "AlephBFT-member", "incorrectly signed unit in a response: {:?}", e);
+                    return;
+                }
+            };
+            if self
+                .store
+                .unit_by_hash(&checked_unit.as_signable().hash())
+                .is_none()
+            {
+                let starting_round_candidate = checked_unit.as_signable().round() + 1;
+                self.on_unit_received(checked_unit.into(), false);
+                if starting_round_candidate > self.starting_round_value {
+                    self.starting_round_value = starting_round_candidate;
+                }
+            }
+        }
+        self.newest_unit_responders.insert(response.responder);
+        if self.is_starting_round_ready() {
+            trace!(target: "AlephBFT-runway", "sending starting round to creator: {}", self.starting_round_value);
```

Comment:
Shall we send to `self.resolved_requests` somewhere here?

Reply:
Yes we should, fixed.